docs/source/API/core/parallel-dispatch/parallel_for.rst

```diff
@@ -0,0 +1,127 @@
+``parallel_for()``
+==================
+
+.. role::cpp(code)
+    :language: cpp
+
+Header File: ``<Kokkos_Core.hpp>``
+
+Usage:
+
+.. code-block:: cpp
+
+    Kokkos::parallel_for(name, policy, functor);
+    Kokkos::parallel_for(policy, functor);
+
+.. _text: ../policies/ExecutionPolicyConcept.html
+
+.. |text| replace:: *ExecutionPolicy*
+
+Dispatches parallel work defined by ``functor`` according to the |text|_ ``policy``. The optional label ``name`` is
+used by profiling and debugging tools. This call may be asynchronous and return to the callee immediately. 
+
+Interface
+---------
+
+.. code-block:: cpp
+
+    template <class ExecPolicy, class FunctorType>
+    Kokkos::parallel_for(const std::string& name, 
+                         const ExecPolicy& policy, 
+                         const FunctorType& functor);
+
+.. code-block:: cpp
+
+    template <class ExecPolicy, class FunctorType>
+    Kokkos::parallel_for(const ExecPolicy& policy, 
+                         const FunctorType& functor);
+
+Parameters:
+~~~~~~~~~~~
+
+* ``name``: A user provided string which is used in profiling and debugging tools via the Kokkos Profiling Hooks. 
+* ExecPolicy: An *ExecutionPolicy* which defines iteration space and other execution properties. Valid policies are:
+
+  - ``IntegerType``: defines a 1D iteration range, starting from 0 and going to a count.
+  - `RangePolicy <../policies/RangePolicy.html>`_: defines a 1D iteration range.
+  - `MDRangePolicy <../policies/MDRangePolicy.html>`_: defines a multi-dimensional iteration space.
+  - `TeamPolicy <../policies/TeamPolicy.html>`_: defines a 1D iteration range, each of which is assigned to a thread team.
```

Comment:
Future work (not required for merging now):  fix expression for `TeamPolicy`; should read something like, "...defines an execution policy for  1D iteration space starting at the begin iterate and going to the end iterate."